src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetInfoService.cs

```diff
@@ -123,64 +123,60 @@ private void PopulateSnippetCaches()
         {
             Debug.Assert(_expansionManager != null);
 
-            // Ideally we'd fork execution here based on whether the expansion manager is an
-            // IExpansionManager or not. Unfortunately, we cannot mention that type by name until
-            // the Roslyn build machines are upgraded to Visual Studio 2015 Update 1. We therefore
-            // need to try using IExpansionManager dynamically, from a background thread. If that
-            // fails, then we come back to the UI thread for using IVsExpansionManager instead.
-
             var token = _waiter.BeginAsyncOperation(GetType().Name + ".Start");
 
-            Task.Factory.StartNew(async () => await PopulateSnippetCacheOnBackgroundWithForegroundFallback().ConfigureAwait(false),
-                            CancellationToken.None,
-                            TaskCreationOptions.None,
-                            TaskScheduler.Default).CompletesAsyncOperation(token);
+            var asyncExpansionManager = _expansionManager as IExpansionManager;
```

Comment:
@jasonmalinowski Yes, in case we need to revert back if we find a bug in production.  The test coverage on this feature isn't very good, so we can't reliably say that it's bug free.  The plan is to release, if we don't hear any complaints for a couple of months, then we can remove.